base/array.jl

```diff
@@ -711,8 +711,8 @@ function find(testf::Function, A::AbstractArray)
     # use a dynamic-length array to store the indexes, then copy to a non-padded
     # array for the return
     tmpI = Array(Int, 0)
-    for i = 1:length(A)
-        if testf(A[i])
+    for (i,j) = enumerate(eachindex(A))
+        if testf(A[j])
```

Comment:
In fact the code seems to have a bug as written. It should be something like
```jl
for (i, a) in enumerate(A)
    if testf(a)
```
(In other words, `a = A[i]`.)